cfme/infrastructure/networking.py

```diff
@@ -1,73 +1,118 @@
-from widgetastic_manageiq import PaginationPane
-from widgetastic_manageiq import Text
+from widgetastic_manageiq import ManageIQTree
+from widgetastic_manageiq import SummaryTable
 
 
-class InfraNetworking(BaseEntity):
-    _param_name = 'InfraNetworking'
+@attr.s
+class InfraSwitches(BaseEntity, Taggable, CustomButtonEventsMixin):
+    """ Model of an infrastructure cluster in cfme
+
+    Args:
+        name: Name of the switch.
+    """
+
+    name = attr.ib()
 
 
 @attr.s
-class InfraNetworkingCollection(BaseCollection):
+class InfraSwitchesCollection(BaseCollection):
     """Collection object for the :py:class:`cmfe.infrastructure.networking.InfraNetworking`."""
-    ENTITY = InfraNetworking
+
+    ENTITY = InfraSwitches
+
+    def all(self):
+        """List of all switch objects"""
+        view = navigate_to(self, "All")
+        return [self.instantiate(ent) for ent in view.entities.entity_names]
 
 
 class InfraNetworkingView(BaseLoggedInPage):
     """Base view for header and nav checking, navigatable views should inherit this"""
 
     @property
     def in_infra_networking(self):
-        nav_chain = ['Compute', 'Infrastructure', 'Networking']
-        return (
-            self.logged_in_as_current_user and
-            self.navigation.currently_selected == nav_chain)
+        nav_chain = ["Compute", "Infrastructure", "Networking"]
+        return self.logged_in_as_current_user and self.navigation.currently_selected == nav_chain
 
 
 class InfraNetworkingToolbar(View):
-    """The toolbar on the main page"""
-    policy = Dropdown('Policy')
+    """The toolbar for Infra Networking view"""
 
+    policy = Dropdown("Policy")
     view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-class InfraNetworkingEntities(View):
-    """Entities on the main page"""
-    title = Text('//div[@id="main-content"]//h1')
+class InfraSwitchesAllView(InfraNetworkingView):
+    """Infra switch all view"""
+
+    @property
+    def is_displayed(self):
+        return self.in_infra_networking and self.entities.title.text == "All Switches"
+
+    toolbar = View.nested(InfraNetworkingToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @View.nested
+    class switches(Accordion):  # noqa
+        ACCORDION_NAME = "Switches"
+        tree = ManageIQTree()
+
+
+class InfraSwitchesDetailsEntities(View):
+    """The entities on the detail page"""
+
+    title = Text('//*[@id="explorer_title_text"]')
+    relationships = SummaryTable("Relationships")
+    smart_management = SummaryTable("Smart Management")
+
 
+class InfraNetworkingDetailsView(InfraNetworkingView):
+    """The details page of a switch"""
 
-class InfraNetworkingAllView(InfraNetworkingView):
-    """The "all" view -- a list"""
     @property
     def is_displayed(self):
-        return (
-            self.in_infra_networking and
-            self.entities.title.text == 'All Switches')
+        """Determine if this page is currently being displayed"""
```

Comment:
I don't think this comment is needed, the name is self-explanatory

Reply:
I think i copied that :)